java/ql/src/semmle/code/java/dataflow/FlowSources.qll

```diff
@@ -304,17 +305,22 @@ class ReverseDNSMethod extends Method {
   }
 }
 
-/** Android `Intent` that may have come from a hostile application. */
-class AndroidIntentInput extends DataFlow::Node {
+/** Exported Android `Intent` that may have come from a hostile application. */
+class AndroidIntentInput extends RemoteFlowSource {
```

Comment:
Let's slightly vary this by keeping `AndroidIntentInput` that just extends `DataFlow::Node` then adding a qualifying class right below it `ExportedAndroidIntentInput` that adds the `isExported()` qualifier; that way any existing query that uses `AndroidIntentInput` to talk about internal communication won't be disturbed

Reply:
Deleted changes to <code>AndroidIntentInput</code> and added <code>ExportedAndroidIntentInput</code>